### ISSUE TITLE
Trying to map a nested field fails

### ISSUE THREAD

**Issue by construe** [JoshClose/CsvHelper#785]
Following code generates InvalidCastException.

```
class A
{
	public int AField { get; set; }
}
class B
{
	public A Nested { get; set; }
}

[TestMethod]
public void NestedFieldMapTest()
{
	var configuration = new CsvHelper.Configuration.Configuration();
	var map = (ClassMap<B>)configuration.AutoMap<B>();
	map.Map( b => b.Nested.AField ).Name( "x" );
}
```

Is this the right use case though?
How can you amend a mapping for a nested field?

**Comment by JoshClose** [comment]
Looks like there is an issue with overriding nested maps that way. I'll look into this.

**Comment by JoshClose** [comment]
I fixed this in version 3.4.0 which is on NuGet. Auto mapping will create the correct type of member map for this to work. It still won't work if you were to do `Reference<AMap>( m => Nested )` in a custom class map though. I don't see a reason to even have `Reference` available any more. I may just remove it.

**Comment by construe** [comment]
Works for me and more convenient now, thank you!
